packages/device_info/device_info_platform_interface/lib/method_channel_device_info.dart

```diff
@@ -0,0 +1,30 @@
+import 'dart:async';
+
+import 'package:flutter/services.dart';
+import 'package:device_info_platform_interface/model/androidDeviceIno.dart';
+import 'package:device_info_platform_interface/model/iosDeviceInfo.dart';
+
+import 'device_info_platform_interface.dart';
+
+const MethodChannel _channel = MethodChannel('plugins.flutter.io/device_info');
+
+/// An implementation of [DeviceInfoPlatform] that uses method channels.
+class MethodChannelDeviceInfo extends DeviceInfoPlatform {
+  // Method channel for Android devices
+  Future<AndroidDeviceInfo> androidInfo() async {
```

Comment:
Can you add a couple of unit tests to this methods? You just need to mock the response of the "invokeCall" (you might need to make the `_channel` public and mark it with `@visibleForTesting`, here's an example of method channel tests: https://github.com/flutter/plugins/blob/master/packages/connectivity/connectivity_platform_interface/test/method_channel_connectivity_test.dart)

Reply:
I'll try. I haven't written unit tests before. I'll try my best for this.

Reply:
@yash1200 you just need to mock the response of the method call, and then assert that what the method of the class is returning matches what you expect with what you mocked. Typical tests are: what happens if the MethodChannel returns a "normal" response (with all the expected fields) and what if the MethodChannel returns `null` (this might make the current method crash!)

I think the connectivity example is a good one for you to get started with unit testing!

Reply:
Unit tests are added now.